### PR TITLE
feature of (Average Fruit And Vegetables) done.

### PR DESCRIPTION
**Pull request by msalarab**

### CODE REVIEW

File: src/main/kotlin/interactor/GetAverageFruitAndVegetablesInteractor.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import model.FruitAndVegetablesPrices
+
+class GetAverageFruitAndVegetablesInteractor(
+    private val dataSource: CostOfLivingDataSource,
+) {
+    fun execute(): List<String> {
+        return dataSource
+            .getAllCitiesData()
+            .filter(::excludeNullAndNonNumberSalaries)
+            .sortedBy { it.fruitAndVegetablesPrices.getAverageFruitVegetablesPrice() / it.averageMonthlyNetSalaryAfterTax!! }
```
Comment (AmeerAmjed) [inline src/main/kotlin/interactor/GetAverageFruitAndVegetablesInteractor.kt]: Relocate this function from its current position to an external function, and modify the "/" operator to utilize the **div()** function instead.

File: src/test/kotlin/interactor/GetAverageFruitAndVegetablesInteractorTest.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package interactor
+
+import dataSource.utilis.FakeData
+import org.junit.jupiter.api.Assertions.*
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class GetAverageFruitAndVegetablesInteractorTest {
```
Comment (AmeerAmjed) [inline src/test/kotlin/interactor/GetAverageFruitAndVegetablesInteractorTest.kt:10]: you need more test case "one not enough" 
one case 
1. return empty list if all cities have no average salary.
try add more

**Review by AmeerAmjed — CHANGES_REQUESTED** [review]

File: src/main/kotlin/interactor/GetAverageFruitAndVegetablesInteractor.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            .take(10)
+            .map { it.cityName }
+    }
+
+    private fun excludeNullAndNonNumberSalaries(city: CityEntity): Boolean {
+        return city.averageMonthlyNetSalaryAfterTax != null && !city.averageMonthlyNetSalaryAfterTax.isNaN()
+    }
+
+    private fun FruitAndVegetablesPrices.getAverageFruitVegetablesPrice(): Double {
+        return with(this) {
```
Comment (mosayed01) [inline src/main/kotlin/interactor/GetAverageFruitAndVegetablesInteractor.kt]: Here you don't need **with** keyword

Comment (mosayed01) [reply src/main/kotlin/interactor/GetAverageFruitAndVegetablesInteractor.kt]: Because your scope in this time already **this**

**Review by mosayed01 — CHANGES_REQUESTED** [review]
Nice work 
Just look at reviews required

File: src/test/kotlin/interactor/GetAverageFruitAndVegetablesInteractorTest.kt [hunk 4]
```diff
@@ -19,6 +19,20 @@ class GetAverageFruitAndVegetablesInteractorTest {
     }
 
     @Test
-    fun execute() {
+    fun should_ReturnListOf10CitiesNames_When_TheDataSourceIsValid() {
```
Comment (AmeerAmjed) [inline src/test/kotlin/interactor/GetAverageFruitAndVegetablesInteractorTest.kt:22]: you should use style **(given , when, then)** and write description

**Review by AmeerAmjed — CHANGES_REQUESTED** [review]

**Pull request closed without merging**